code-challenges.test.js

```diff
@@ -14,38 +14,114 @@
+  })
+}
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+describe('remainders', () => {
+  it('takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3', () => {
+    const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+    const hodgepodge1Output = [2, 0, -1, 0]
+    // Expected output: [ 2, 0, -1, 0 ]
+
+    const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+    const hodgepodge2Output = [2, 1, -1]
+    // Expected output: [ 2, 1, -1 ]
 
+    expect(remainders(hodgepodge1)).toEqual(hodgepodge1Output)
+    expect(remainders(hodgepodge2)).toEqual(hodgepodge2Output)
+  })
+})
+
+// ● remainders › takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3
+// ReferenceError: remainders is not defined
 
 // b) Create the function that makes the test pass.
 
+// create a function named remainders
+// remainders will take in an array of mixed data
+// we will have to filter anything in the array that is not a number
+// we can loop over the filtered array and make a new array containing the remainder of that number divided by 3
+// we can then return the new array
+
+const remainders = arr => arr.filter(val => typeof val === 'number').map(val => val % 3)
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+describe('cubedSum', () => {
+  it('takes in an array of numbers and returns the sum of all the numbers cubed', () => {
+    const cubeAndSum1 = [2, 3, 4]
+    const cubeAndSum1Output = 99
+    // Expected output: 99
+    const cubeAndSum2 = [0, 5, 10]
+    const cubeAndSum2Output = 1125
+    // Expected output: 1125
 
+    expect(cubedSum(cubeAndSum1)).toEqual(cubeAndSum1Output)
+    expect(cubedSum(cubeAndSum2)).toEqual(cubeAndSum2Output)
+  })
+})
+
+// ● cubedSum › takes in an array of numbers and returns the sum of all the numbers cubed
+// ReferenceError: cubedSum is not defined
 
 // b) Create the function that makes the test pass.
+
+// create a function named cubedSum
+// cubedSum will take in an array of numbers
+// we can loop over the array and cube each number and add it to a new array
+// we can then loop over that array and find the sum of all the cubed numbers
+// we can then return the sum
+
+const cubedSum = arr => arr.map(val => val ** 3).reduce((a, b) => a + b)
```

Comment:
Everything looks terrific. You know the one-liners are special to me. Please include your test suite results after all the functions are created. I feel that the functions passed. I just need to see that the test suite feels the same way.

Reply:
I'm sorry, that's my mistake. I have included the test results after each function creation as per request. Thank you all the feedback on my assessments over the weeks. I appreciate you Charlean!